src/uct/ib/rc/base/rc_iface.c

```diff
@@ -150,39 +150,52 @@ ucs_status_t uct_rc_iface_query(uct_rc_iface_t *iface,
                                   UCT_IFACE_FLAG_EVENT_SEND_COMP |
                                   UCT_IFACE_FLAG_EVENT_RECV;
 
-    if (dev->atomic_arg_sizes & sizeof(uint64_t)) {
-        /* TODO: remove deprecated flags */
-        iface_attr->cap.flags              |= UCT_IFACE_FLAG_ATOMIC_DEVICE;
+    if (dev->pci_fadd_arg_sizes || dev->pci_cswap_arg_sizes) {
+        if (dev->pci_fadd_arg_sizes & sizeof(uint64_t)) {
+            iface_attr->cap.atomic64.op_flags  |= UCS_BIT(UCT_ATOMIC_OP_ADD);
+            iface_attr->cap.atomic64.fop_flags |= UCS_BIT(UCT_ATOMIC_OP_ADD);
+        }
+        if (dev->pci_cswap_arg_sizes & sizeof(uint64_t)) {
+            iface_attr->cap.atomic64.fop_flags |= UCS_BIT(UCT_ATOMIC_OP_CSWAP);
+        }
+#if HAVE_IB_EXT_ATOMICS
+        if (dev->pci_fadd_arg_sizes & sizeof(uint32_t)) {
+            iface_attr->cap.atomic32.op_flags  |= UCS_BIT(UCT_ATOMIC_OP_ADD);
+            iface_attr->cap.atomic32.fop_flags |= UCS_BIT(UCT_ATOMIC_OP_ADD);
+        }
+        if (dev->pci_cswap_arg_sizes & sizeof(uint32_t)) {
+            iface_attr->cap.atomic32.fop_flags |= UCS_BIT(UCT_ATOMIC_OP_CSWAP);
+        }
+#endif
+        iface_attr->cap.flags                  |= UCT_IFACE_FLAG_ATOMIC_CPU;
+    } else {
+        if (dev->atomic_arg_sizes & sizeof(uint64_t)) {
+            /* TODO: remove deprecated flags */
+            iface_attr->cap.flags              |= UCT_IFACE_FLAG_ATOMIC_DEVICE;
```

Comment:
why is this inside 'if'? should be right before line 172, in the 'else' clause of `if (dev->pci_fadd_arg_sizes || dev->pci_cswap_arg_sizes) {`